addons/account/models/chart_template.py

```diff
@@ -565,15 +565,23 @@ def _load_template(self, company, code_digits=None, account_ref=None, taxes_ref=
 
     @api.multi
     def create_record_with_xmlid(self, company, template, model, vals):
-        # Create a record for the given model with the given vals and
-        # also create an entry in ir_model_data to have an xmlid for the newly created record
-        # xmlid is the concatenation of company_id and template_xml_id
-        ir_model_data = self.env['ir.model.data']
-        template_xmlid = ir_model_data.search([('model', '=', template._name), ('res_id', '=', template.id)])
-        xml_id = "%s.%s_%s" % (template_xmlid.module, company.id, template_xmlid.name)
-        data = dict(xml_id=xml_id, values=vals, noupdate=True)
-        record = self.env[model]._load_records([data])
-        return record.id
+        return self._create_records_with_xmlid(model, [(template, vals)], company).id
+
+    def _create_records_with_xmlid(self, model, template_vals, company):
+        """ Create records for the given model name with the given vals, and
+            create xml ids based on each record's template and company id.
+        """
+        if not template_vals:
+            return self.env[model]
+        template_model = template_vals[0][0]
+        template_ids = [template.id for template, vals in template_vals]
+        template_xmlids = template_model.browse(template_ids).get_external_id()
```

Comment:
I did it that way because the method `get_external_id` is made for that purpose.
It avoids having to know the details of `ir.model.data` to understand your own code.